src/pipx/venv.py

```diff
@@ -173,21 +176,21 @@ def install_package(
         is_main_package: bool,
     ) -> None:
 
-        with animate(f"installing package {package_or_url!r}", self.do_animation):
-            if pip_args is None:
-                pip_args = []
-            if package is None:
-                # If no package name is supplied, install only main package
-                #   first in order to see what its name is
-                old_package_set = self.list_installed_packages()
-                cmd = ["install"] + pip_args + ["--no-dependencies"] + [package_or_url]
-                self._run_pip(cmd)
-                installed_packages = self.list_installed_packages() - old_package_set
-                if len(installed_packages) == 1:
-                    package = installed_packages.pop()
-                    logging.info(f"Determined package name: '{package}'")
-                else:
-                    package = None
+        if pip_args is None:
+            pip_args = []
+        if package is None:
+            # If no package name is supplied, install only main package
+            #   first in order to see what its name is
+            package = self.install_package_no_deps(package_or_url, pip_args)
+
+        install_message = f"installing {package}"
+        if package != package_or_url:
+            install_message += f" from specification {package_or_url!r}"
+
```

Comment:
Looks like this can be replaced with a call to `full_package_description()`?

Reply:
Oops, actually those three lines can be deleted, I replaced install_message with `full_package_description()` in animate later on!